src/data.ts

```diff
@@ -64,6 +90,98 @@ export class PnpmWorkspaceManager {
         disposable.dispose()
       }
     })
+
+    return data
+  }
+
+  private readPnpmWorkspacePosition(doc: TextDocument) {
+    if (this.positionDataMap.has(doc.uri.fsPath)) {
+      return this.positionDataMap.get(doc.uri.fsPath)!
+    }
+
+    let defaultNamePosition: [number, number] = [0, 0]
+    let namedStartPosition: [number, number] = [0, 0]
+    let custimzedNamePosition: [number, number] = [0, 0]
+
+    let data: PnpmWorkspacPositionData = {
+      catalog: undefined,
+      catalogs: undefined
+    }
+
+    let currentCatalogName = ''
+    for (let lineIndex = 0; lineIndex < doc.lineCount; lineIndex++) {
```

Comment:
Should we use a proper YAML parser instead of regex?

Reply:
I haven't found a suitable library; the one I'm currently using, js-yaml, doesn't support it

Reply:
Maybe try https://github.com/ota-meshi/yaml-eslint-parser?

Reply:
Thank you, I will look into it later.